bin/oneoff/backfill_parent-child_relationships.rb

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ruby
+require_relative '../../lib/cdo/only_one'
+abort 'Script already running' unless only_one_running?(__FILE__)
+
+require_relative '../../dashboard/config/environment'
+
+def main
+  Level.includes(:child_levels).all.each do |level|
+    if level.contained_level_names.present?
+      begin
+        level.setup_contained_levels
```

Comment:
I thought this migration was already performed with https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/db/migrate/20210722193724_setup_contained_levels.rb, I'm curious why this is necessary.

Reply:
Good catch! Yes, this migration was run on all environments with that change, but there was some bad data in a couple of our environments and also a small bug, both of which mean that there is a subset of levels that didn't get correctly updated by that. So this script is pulling double duty; both migrating over the project template levels for the first time and finishing the migration for the contained levels.